seed/seed20_test.go

```
@@ -2873,3 +2876,178 @@ func (s *seed20Suite) TestLoadMetaWrongHashSnapParallelism2(c *C) {
+	c.Assert(err, IsNil)
+
+	return seed20
+}
+
+var goodUser = map[string]interface{}{
+	"authority-id": "my-brand",
+	"brand-id":     "my-brand",
+	"email":        "foo@bar.com",
+	"series":       []interface{}{"16", "18"},
+	"models":       []interface{}{"my-model", "other-model"},
+	"name":         "Boring Guy",
+	"username":     "guy",
+	"password":     "$6$salt$hash",
+	"since":        time.Now().Format(time.RFC3339),
+	"until":        time.Now().Add(24 * 30 * time.Hour).Format(time.RFC3339),
+}
+
+func (s *seed20Suite) writeValidAutoImportAssertion(c *C, sysLabel string, perm os.FileMode) {
+	systemUsers := []map[string]interface{}{goodUser}
+	// write system user assertion to the system seed root
+	autoImportAssert := filepath.Join(s.SeedDir, "systems", sysLabel, "auto-import.assert")
+	f, err := os.OpenFile(autoImportAssert, os.O_CREATE|os.O_WRONLY, perm)
+	c.Assert(err, IsNil)
+	defer f.Close()
+	enc := asserts.NewEncoder(f)
+	c.Assert(enc, NotNil)
+
+	for _, suMap := range systemUsers {
+		systemUser, err := s.Brands.Signing(suMap["authority-id"].(string)).Sign(asserts.SystemUserType, suMap, nil, "")
+		c.Assert(err, IsNil)
+		systemUser = systemUser.(*asserts.SystemUser)
+		err = enc.Encode(systemUser)
+		c.Assert(err, IsNil)
+	}
+}
+
+func (s *seed20Suite) writeInvalidAutoImportAssertion(c *C, sysLabel string, perm os.FileMode) {
+	autoImportAssert := filepath.Join(s.SeedDir, "systems", sysLabel, "auto-import.assert")
+	// write random data
+	randomness := make([]byte, 512)
+	rand.Read(randomness)
+	err := ioutil.WriteFile(autoImportAssert, randomness, perm)
+	c.Assert(err, IsNil)
+}
+
+// findAutoImportAssertion returns found systemUser assertion
+func (s *seed20Suite) findAutoImportAssertion(seed20 seed.Seed) ([]asserts.Assertion, error) {
+	assertions, err := s.db.FindMany(asserts.SystemUserType, map[string]string{
+		"brand-id": seed20.Model().BrandID(),
+	})
+
+	if asserts.IsNotFound(err) {
+		return nil, fmt.Errorf("No system user assertion found")
```

this error conversion is a bit strange, I would simply return the error as they are and have the callers deal with them directly

---

Indeed. I have removed the condition and returning the error.